base/collect_grpc.go

```diff
@@ -64,25 +72,72 @@ func (t *collectGRPCTask) validateInputs() error {
 }
 
 // resultForVersion collects gRPC test result for a given version
-func (t *collectGRPCTask) resultForVersion() (*runner.Report, error) {
+func (t *collectGRPCTask) resultForVersion() (map[string]*runner.Report, error) {
 	// the main idea is to run ghz with proper options
 
-	opts := runner.WithConfig(&t.With.Config)
+	var err error
+	results := map[string]*runner.Report{}
+
+	if len(t.With.Endpoints) > 0 {
+		log.Logger.Trace("multiple endpoints")
+		for endpointID, endpoint := range t.With.Endpoints {
+			endpoint := endpoint // prevent implicit memory aliasing
+			log.Logger.Trace(fmt.Sprintf("endpoint: %s", endpointID))
+
+			// default from baseline
+			call := t.With.Call
+			if endpoint.Call != "" {
+				call = endpoint.Call
+			}
+
+			host := t.With.Host
+			if endpoint.Host != "" {
+				host = endpoint.Host
+			}
 
-	// todo: supply all the allowed options
-	igr, err := runner.Run(t.With.Call, t.With.Host, opts)
-	if err != nil {
-		e := errors.New("ghz run failed")
-		log.Logger.WithStackTrace(err.Error()).Error(e)
-		if igr == nil {
-			e = errors.New("failed to get results since ghz run was aborted")
-			log.Logger.Error(e)
+			// merge endpoint options with baseline options
+			if err := mergo.Merge(&endpoint, t.With.Config); err != nil {
+				log.Logger.Error(fmt.Sprintf("could not merge Fortio options for endpoint \"%s\"", endpointID))
+				return nil, err
+			}
+			eOpts := runner.WithConfig(&endpoint) // endpoint options
+
+			igr, err := runner.Run(call, host, eOpts)
```

Comment:
Are we accessing the endpoints sequentially? Is this what we want?  Or do we want this to be concurrent executions? I think this would be my expectation.

Reply:
This is an optimization. Definitely, a useful optimization... but perhaps, one that can handled in a future PR? For now, sequential seems like a good (MVP) start.